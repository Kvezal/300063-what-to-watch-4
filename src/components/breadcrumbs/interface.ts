export interface IBreadcrumb {
  name: string;
  href?: string;
}

export interface IBreadcrumbsProps {
  breadcrumbs: IBreadcrumb[];
}
