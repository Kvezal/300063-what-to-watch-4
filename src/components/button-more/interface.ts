interface IButtonMoreProps {
  hide: boolean;
  onButtonClick: () => void;
}


export {
  IButtonMoreProps,
};
