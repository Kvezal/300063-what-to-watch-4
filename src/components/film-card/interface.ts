import {ReactNode} from "react";


export interface IFilmCardProps {
  filmId: number;
  filmName: string;
  onCardClick: (filmId: number) => void;
  onPlayingChange: (isPlaying: boolean) => void;
  renderPlayer: () => ReactNode;
}
