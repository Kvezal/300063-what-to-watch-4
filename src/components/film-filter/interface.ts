export interface IFilmFilterProps {
  tabs: string[];
  onItemClick: () => void;
  baseURI: string;
  activeItem: string;
}
