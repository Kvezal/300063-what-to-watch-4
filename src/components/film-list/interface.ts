import {FilmInterface} from "@common/types";


interface IFilmListProps {
  list: FilmInterface[];
  onCardClick: (filmId: number) => void;
  pack?: number;
  step?: number;
}

export {
  IFilmListProps,
};
