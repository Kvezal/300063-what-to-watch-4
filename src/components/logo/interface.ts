export interface ILogoProps {
  linkClass?: {
    [key: string]: boolean;
  };
}
