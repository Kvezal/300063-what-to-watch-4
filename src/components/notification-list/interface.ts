import {INotification} from "@store/notification/interface";


export interface INotificationList {
  notifications: INotification[];
  onCloseNotification: (notificationId: string) => void;
  maxItems: number;
}
