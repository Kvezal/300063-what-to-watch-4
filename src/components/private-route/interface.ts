import * as React from "react";
import {RouteProps} from "react-router-dom";


interface IPrivateRoute {
  path: string;
  exact: boolean;
  render: (props: RouteProps) => React.ReactNode;
  authorizationStatus: string;
}

type TPrivateRoute = RouteProps & IPrivateRoute;

export {
  IPrivateRoute,
  TPrivateRoute,
};
