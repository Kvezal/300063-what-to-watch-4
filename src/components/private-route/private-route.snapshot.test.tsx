import * as React from "react";
import * as render from "react-test-renderer";
import {MemoryRouter, Route} from "react-router-dom";
import {EAuthorizationStatus} from "@store/user/interface";

import PrivateRoute from "./private-route";


describe(`PrivateRouterSnapshot`, () => {
  test(`should render private component`, () => {
    const tree = render.create(
        <MemoryRouter initialEntries={[`/private`]}>
          <PrivateRoute
            path="/private"
            exact
            authorizationStatus={EAuthorizationStatus.AUTH}
            redirect="/not-private"
            render={() => <div className="test"/>}
          />
          <Route
            path="/login"
            render={() => <div className="not-private-route"/>}
          />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`shouldn't render private component`, () => {
    const tree = render.create(
        <MemoryRouter initialEntries={[`/private`]}>
          <PrivateRoute
            path="/private"
            exact
            authorizationStatus={EAuthorizationStatus.NO_AUTH}
            redirect="/not-private"
            render={() => <div className="test"/>}
          />
          <Route
            path="/login"
            render={() => <div className="not-private-route"/>}
          />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

