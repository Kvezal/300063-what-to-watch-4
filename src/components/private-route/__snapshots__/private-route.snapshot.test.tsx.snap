// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouterSnapshot should render private component 1`] = `
<div
  className="test"
/>
`;

exports[`PrivateRouterSnapshot shouldn't render private component 1`] = `
<div
  className="not-private-route"
/>
`;
